test/test_jit.py

```diff
@@ -7396,6 +7396,18 @@ def any_refinement2(a):
         self.assertEqual(any_refinement2(3), torch.tensor(3))
         self.assertEqual(any_refinement2(torch.tensor(5)), torch.tensor(5))
 
+    def test_unspecialized_any_binding(self):
+        # any binding will infer the type, if it infers
+        # a specialized tensor type `x` Dict type will fail isinstance check
+
+        @torch.jit.script
+        def foo(x: Any):
+            assert isinstance(x, Dict[str, torch.Tensor])
```

Comment:
Is there a way at all to express a more specified Tensor type in Python? If so, does this PR break checking against that?

Reply:
No, there's not